src/Libraries/DSOffice/OpenXmlHelper.cs

```diff
@@ -0,0 +1,647 @@
+        {
+            var sheet = document.WorkbookPart.Workbook.GetFirstChild<Sheets>().Elements<Sheet>().FirstOrDefault(s => s.Name == sheetName);
+            if (sheet == null)
+            {
+                return null;
+            }
+            return document.WorkbookPart.GetPartById(sheet.Id.Value) as WorksheetPart;
+        }
+
+        /// <summary>
+        /// Checks whether the passed in data contains a function.
+        /// </summary>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        private static bool ContainsFunction(object[][] data)
+        {
+            foreach (var row in data)
+            {
+                if (row == null)
+                {
+                    continue;
+                }
+
+                foreach (var value in row)
+                {
+                    if (value is ProtoCore.DSASM.StackValue stackValue && stackValue.IsPointer)
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Opens an existing workbook or creates one from scratch on the specified file path.
+        /// </summary>
+        /// <param name="filePath">File path where the workbook is or should be created</param>
+        /// <param name="sheetName">In case a new workbook is created, the sheet will use this name</param>
+        /// <returns>A spreadsheet document</returns>
+        private static SpreadsheetDocument OpenOrCreate(string filePath, string sheetName)
+        {
+            if (File.Exists(filePath))
+            {
+                return SpreadsheetDocument.Open(filePath, true);
+            }
+            else
+            {
+                var document = SpreadsheetDocument.Create(filePath, SpreadsheetDocumentType.Workbook);
+                var workbookPart = document.AddWorkbookPart();
+                workbookPart.Workbook = new Workbook();
+                var worksheetPart = workbookPart.AddNewPart<WorksheetPart>();
+                worksheetPart.Worksheet = new Worksheet(new SheetData());
+                var sheets = workbookPart.Workbook.AppendChild(new Sheets());
+                var sheet = new Sheet()
+                {
+                    Id = workbookPart.GetIdOfPart(worksheetPart),
+                    SheetId = 1,
+                    Name = sheetName
+                };
+                sheets.Append(sheet);
+                
+                var workbookStylesPart = workbookPart.AddNewPart<WorkbookStylesPart>();
+                workbookStylesPart.Stylesheet = new Stylesheet();
+                AddDefaultStyles(workbookStylesPart.Stylesheet);
+                return document;
+            }
+        }
+
+        /// <summary>
+        /// Adds default styles to the stylesheet so that Office does not identify the spreadsheet as corrupt.
```

Comment:
Is there any reference to the changes in this function? Do you foresee MS change how they identify spreadsheet as corrupt?

Reply:
That was something I noticed in practice when the styles were absent, even if they were not needed. This code is the result of reviewing the styles of a newly created spreadsheet in Excel and adding some of those elements until Excel would accept the file created by Dynamo. Unfortunately, there is no documentation for this AFAIK, so the process was pretty hard.

Reply:
@mmisol I can only imagine... Well, that is a rough start and I hope MS dont change these criteria often, otherwise we are in big trouble..